src/server/campaignd/server.cpp

```diff
@@ -1143,9 +1164,11 @@ void server::handle_request_campaign(const server::request& req)
 		}
 
 		LOG_CS << req << "Sending add-on '" << name << "' version: " << to << " size: " << full_pack_size / 1024 << " KiB\n";
-		boost::system::error_code ec;
-		coro_send_file(req.sock, full_pack_path, req.yield[ec]);
-		if(check_error(ec, req.sock)) return;
+		utils::visit([this, &req, &full_pack_path](auto&& socket) {
+			boost::system::error_code ec;
+			coro_send_file(socket, full_pack_path, req.yield[ec]);
+			if(check_error(ec, socket)) return;
```

Comment:
Why a return statement at the end of the function?

Reply:
Because it was standard pattern in code before? But yeah it probably indicates some confusion here. Actually I'm not even sure about the best way to make those visitors interrupt calling function.

Reply:
If I throw instead then exception will actually originate from asio's event loop and would need additional handling to prevent them from stopping the loop.

Reply:
I fixed those cases of unintentionally ignoring errors. Though I'll most likely eventually make error handling everywhere exception based.